pyswarms/utils/decorators/decorators.py

```diff
@@ -39,9 +40,15 @@ def cost_func(x):
     cost_dec : callable
         The vectorized output for all particles as defined by :code:`cost_func`
     """
+
     def cost_dec(particles, **kwargs):
         n_particles = particles.shape[0]
-        vector = np.array([cost_func(particles[i], **kwargs) for i in range(n_particles)])
-        assert vector.shape == (n_particles, ), "The cost function should return a single value."
+        vector = np.array(
+            [cost_func(particles[i], **kwargs) for i in range(n_particles)]
+        )
+        assert vector.shape == (
+            n_particles,
+        ), "The cost function should return a single value."
```

Comment:
Maybe we should remove this assert statement 🤔 We could replace it with a try-except.

Reply:
Yes you’re correct, but this is more of a validator that ensures vector.shape == (n_particles,), maybe we can do a LBYL style here? ( Let’s just remove asserts since assert statements can be skipped during runtime) 👍